pylint/testutils/functional/test_file.py

```diff
@@ -0,0 +1,75 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+
+import configparser
+from os.path import basename, exists, join
+
+
+def parse_python_version(ver_str):
+    return tuple(int(digit) for digit in ver_str.split("."))
+
+
+class NoFileError(Exception):
+    pass
+
+
+class FunctionalTestFile:
+    """A single functional test case file with options."""
+
+    _CONVERTERS = {
+        "min_pyver": parse_python_version,
+        "max_pyver": parse_python_version,
+        "min_pyver_end_position": parse_python_version,
+        "requires": lambda s: s.split(","),
+    }
+
+    def __init__(self, directory, filename):
```

Comment:
Shall we try and type this while we're looking at it?

Reply:
I started to add str and str for ``FunctionalTestFile`` and this created 8 mypy issues in LintModuleTest because it started to be considered typed, I think this is work for another MR. It's not everywhere so I might be able to add typing in some place.

Reply:
97940f0 (Turns out it's everywhere else, only the constructor was problematic)